lib/ui/survey_question/survey_answer_views/multiple_choice_view.dart

```diff
@@ -0,0 +1,142 @@
+import 'package:flutter/material.dart';
+import 'package:survey_flutter_ic/ui/survey_question/ui_models/multiple_choice_option_ui_model.dart';
+import 'package:survey_flutter_ic/utils/context_ext.dart';
+import 'package:survey_flutter_ic/utils/dimension.dart';
+
+class MultipleChoiceView extends StatefulWidget {
+  final List<MultipleChoiceOptionUIModel> uiModels;
+  final Function(List<String>) onSelect;
+
+  const MultipleChoiceView({
+    super.key,
+    required this.uiModels,
+    required this.onSelect,
+  });
+
+  @override
+  State<MultipleChoiceView> createState() => _MultipleChoiceViewState();
+}
+
+class _MultipleChoiceViewState extends State<MultipleChoiceView> {
+  List<int> selectedIndexes = [];
+
+  Widget get _listView => ListView.separated(
+        itemCount: widget.uiModels.length,
+        itemBuilder: (_, index) => _itemBuilder(
+          uiModels: widget.uiModels,
+          index: index,
+          selectedIndexes: selectedIndexes,
+        ),
+        separatorBuilder: (_, index) => _separatorBuilder(
+          count: widget.uiModels.length,
+          index: index,
+        ),
+      );
+
+  Widget _itemBuilder({
+    required List<MultipleChoiceOptionUIModel> uiModels,
+    required int index,
+    required List<int> selectedIndexes,
+  }) {
+    final uiModel = MultipleChoiceOptionUIModel(
+      id: uiModels[index].id,
+      title: uiModels[index].title,
+      isSelected: selectedIndexes.contains(index),
+    );
+    return GestureDetector(
+      onTap: () => _onTap(index),
+      child: _listItem(uiModel),
+    );
+  }
+
+  Widget _listItem(MultipleChoiceOptionUIModel uiModel) => SizedBox(
+        width: context.screenSize.width - 160.0,
+        height: Dimensions.multipleChoiceItemHeight,
+        child: Row(
+          children: [
+            Expanded(
+              child: Text(
```

Comment:
Please extract the text widget into a class variable:

```suggestion
            Expanded(child: _titleText(context)),
```

```dart
    Widget  _titleText(BuildeContext context) => Text(...
```